internal/format/view.go

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0.
+
+package format
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/mitchellh/colorstring"
+	"github.com/mitchellh/go-wordwrap"
+
+	config "terraform-provider-tfmigrate/internal/configs"
+	"terraform-provider-tfmigrate/internal/diagnostics"
+)
+
+type View interface {
```

Comment:
I don't see where this is configured or created yet, but I see this is built inside a provider? Traditionally, the provider wouldn't display things themselves, and just return diagnostics back to the caller / Terraform to be rendered.

You can see the other existing resources do something similar: https://github.com/hashicorp/terraform-provider-tfmigrate/blob/main/internal/provider/terraform_plan_resource.go#L53. You might not need the views package if you can figure out the best way to package diagnostics up in the same way.